takari-lifecycle-plugin/src/main/java/io/takari/maven/plugins/Deploy.java

```diff
@@ -82,10 +93,15 @@ private void installProject(MavenProject project) throws MojoExecutionException
 
     deployRequest.setRepository(remoteRepository(project));
 
-    try {
-      repositorySystem.deploy(repositorySystemSession, deployRequest);
-    } catch (DeploymentException e) {
-      throw new MojoExecutionException(e.getMessage(), e);
+    if (!deployAtEnd) {
+      try {
+        repositorySystem.deploy(repositorySystemSession, deployRequest);
+      } catch (DeploymentException e) {
+        throw new MojoExecutionException(e.getMessage(), e);
+      }
+    } else {
+      DeployParticipant.deployAtEnd(mavenSession, deployRequest);
```

Comment:
I think it is cleaner to "@Inject DeployParticipant deployParticipant` and use deployParticipant instance instead of static.